src/python/pants/backend/docker/goals/tailor.py

```diff
@@ -34,12 +34,13 @@ async def find_putative_targets(
     pts = []
     for dockerfile in sorted(unowned_dockerfiles):
         dirname, filename = os.path.split(dockerfile)
-        kwargs = {}
+        target_name = "docker"
```

Comment:
Will this properly handle a collision for multiple dockerfiles in one directory?

Reply:
Oh I don't think the kwarg hard coding will work? I'll take a holistic look when I'm back from dentist at how to make the `name` field less error-prone. 

Let's wait to merge this, even if it means we miss the 2.9.0rc0 release we can cherry-pick this. I'd rather fix it the right way.

Reply:
I had this in a PR... https://github.com/pantsbuild/pants/pull/13601#discussion_r748610839

https://github.com/pantsbuild/pants/pull/13601/files/7edf75c30f5bafea9bbe0b8131cc236de0645419#diff-7cb3cfc221bd8a9ef128c86b362e91effd0ecd8d00be5a56d805bf8fe3a47843R40-R45

Think it may be close to what we want here...